### PR TITLE
docs: ✏️ add a section about the search feature

### PR DESCRIPTION
**Pull request by severo**
I also updated the screenshot to include the search bar

### CODE REVIEW

File: docs/hub/datasets-viewer.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ The dataset page includes a table with the contents of the dataset, arranged by
 <img class="hidden dark:block" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/dataset-viewer-dark.png"/>
 </div>
 
+## Search a word in the dataset
+
+You can search a word in the dataset by typing it in the search bar at the top of the table. The search is case-insensitive and will match any row containing the word. The text is searched in the columns of type `string` or `ClassLabel`, even if the values are nested in a dictionary.
```
Comment (davanstrien) [inline docs/hub/datasets-viewer.md:12]:
```suggestion
You can search for a word in the dataset by typing it in the search bar at the top of the table. The search is case-insensitive and will match any row containing the word. The text is searched in the columns of type `string` or `ClassLabel`, even if the values are nested in a dictionary.
```

Comment (davanstrien) [reply docs/hub/datasets-viewer.md:12]: Think the current wording is fine too but slightly prefer this

**Review by davanstrien — APPROVED** [review]
Super cool feature!

**Review by julien-c — COMMENTED** [review]
lgtm

**Pull request merged**